impl/src/main/java/org/ehcache/impl/internal/store/tiering/TieredStore.java

```diff
@@ -299,6 +293,17 @@ private void swapBackCachingTiers() {
     return cachingTierRef.get();
   }
 
+  private ValueHolder<V> handleStoreAccessException(StoreAccessException ce) throws StoreAccessException {
+    Throwable cause = ce.getCause();
+    if (cause instanceof StorePassThroughException) {
+      throw (StoreAccessException) cause.getCause();
+    }
+    if (cause instanceof Error) {
+      throw (Error) cause;
+    }
+    throw (RuntimeException) cause;
+  }
+
```

Comment:
This still isn't safe... checked versus runtime exceptions are a compile time concept only.  I think you'll need somthing like this no?

```java
Throwable cause = ce.getCause();
if (cause instanceof StorePassThroughException) {
  throw (StoreAccessException) cause.getCause();
}
if (cause instanceof Error) {
  throw (Error) cause;
} else if (cause instanceof RuntimeException) {
  throw (RuntimeException) cause;
} else {
  throw new RuntimeException(cause);
}
```

Reply:
I could. But since only runtimes and errors are supposed to be wrapped in the StoreAccessException, anything else means something weird is happening.

Reply:
Let me do something

Reply:
@chrisdennis Done. Please have a look

Reply:
Oups... no... not done.

Reply:
@chrisdennis All good now